code/game/objects/items/tools/flame_tools.dm

```diff
@@ -143,24 +151,32 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 	set_light_on(toggle_on)
 
-/obj/item/tool/match/proc/light_match()
+/obj/item/tool/match/proc/light_match(mob/user)
 	if(heat_source || burnt)
 		return
 	heat_source = 1000
 	playsound(src.loc,"match",15, 1, 3)
 	damtype = "burn"
 	icon_state = "[initial(icon_state)]_lit"
+	item_state = "[initial(item_state)]_lit"
+	if(user)
+		user.update_inv_l_hand()
+		user.update_inv_r_hand()
 	turn_light(toggle_on = TRUE)
 	START_PROCESSING(SSobj, src)
 	update_icon()
 	return TRUE
 
-/obj/item/tool/match/proc/burn_out(mob/user)
+/obj/item/tool/match/proc/burn_out()
 	heat_source = 0
 	burnt = 1
 	damtype = "brute"
 	icon_state = "[initial(icon_state)]_burnt"
-	item_state = "cigoff"
+	item_state = "[initial(item_state)]_burnt"
+	var/mob/holder = loc
+	if(holder)
+		holder.update_inv_l_hand()
+		holder.update_inv_r_hand()
```

Comment:
you have no check that the holder here is a mob. this could also just be burning out on the floor

Reply:
and chaser 
![image](https://github.com/user-attachments/assets/9461de7f-1a48-4d8e-8f4b-70dd89f8cfd5)